frontend/apps/picasso/pages/crowdloan-rewards/index.tsx

```diff
@@ -32,14 +36,17 @@ const CrowdloanRewards: NextPage = () => {
   const { isActive } = useConnector(ConnectorType.MetaMask);
 
   useEffect(() => {
-    if (userAssociation) {
-      userAssociation === "ethereum"
-        ? router.push("crowdloan-rewards/stablecoin")
-        : router.push("crowdloan-rewards/ksm");
+    if (account || selectedAccount) {
+      if (account && account.toLowerCase() in ethereumContributions) {
+        router.push("crowdloan-rewards/claim");
+      }
+
+      if (selectedAccount && encodeAddress(selectedAccount.address, SUBSTRATE_NETWORKS.kusama.ss58Format) in kusamaContributions) {
+        router.push("crowdloan-rewards/claim");
+      }
```

Comment:
We can inverse the condition here to reduce nesting levels:

```suggestion
   if (!account && !selectedAccount) return;  
   
   if (account && account.toLowerCase() in ethereumContributions) {
        router.push("crowdloan-rewards/claim");
   }
   
   if (selectedAccount && encodeAddress(selectedAccount.address, SUBSTRATE_NETWORKS.kusama.ss58Format) in kusamaContributions) {
        router.push("crowdloan-rewards/claim");
   }
```